### PR TITLE
startup: Use $XDG_CONFIG_DIRS/nvim/sysinit.vim if it exists

### LINKED ISSUE #8994: system-wide configuration file
- `nvim --version`:

```
[~]$ nvim --version
NVIM v0.3.1
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wconversion -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -Wno-array-bounds -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.3.1/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info

[~]$ nvim -e --cmd 'echo $VIM|quit'
/usr/share/nvim

[~]$ nvim -e --cmd 'echo $VIMRUNTIME|quit'
/usr/share/nvim/runtime
```

- Vim (version: 8.1) behaves differently?

Yes, vim has `/etc/vimrc`.

- Operating system/version:

Arch Linux

- Terminal name/version:

Konsole

- `$TERM`:

`xterm-256color`

### Steps to reproduce

No system-wide configuration file is used from `/etc`. Instead, the one in `/usr` is used.

### Actual behaviour

nvim reads `/usr/share/nvim/sysinit.vim` as a system-wide configuration file, but this is a vendor-provided file, thus it shouldn't be touched by the user.

### Expected behaviour

nvim should read a system-wide configuration file from `$XDG_CONFIG_DIRS/nvim/sysinit.vim` so the user can customise it.

### PR DESCRIPTION
**Pull request by jamessan**
Closes #8994

### CODE REVIEW

File: src/nvim/main.c [hunk 1]
```diff
@@ -1715,6 +1715,48 @@ static void exe_commands(mparm_T *parmp)
...
+        xfree(config_dirs);
+        return;
+      }
+      xfree(vimrc);
+    } while (iter != NULL);
+  }
+
+#ifdef SYS_VIMRC_FILE
+  // Get system wide defaults, if the file name is defined.
+  (void) do_source((char_u *)SYS_VIMRC_FILE, false, DOSO_NONE);
```
Comment (justinmk) [inline src/nvim/main.c]: Help doc says "Otherwise", but this looks unconditional.

Comment (jamessan) [reply src/nvim/main.c]: The function returns if it successfully loads from `$XDG_CONFIG_DIRS`.

File: src/nvim/main.c [hunk 2]
```diff
@@ -1715,6 +1715,49 @@ static void exe_commands(mparm_T *parmp)
...
+      xfree(vimrc);
+    } while (iter != NULL);
+    xfree(config_dirs);
+  }
+
+#ifdef SYS_VIMRC_FILE
+  // Get system wide defaults, if the file name is defined.
+  (void)do_source((char_u *)SYS_VIMRC_FILE, false, DOSO_NONE);
+#endif
+  return;
```
Comment (justinmk) [inline src/nvim/main.c]: unnecessary return

**Comment by jamessan** [comment]
I should probably add some tests too.  I had started on that, but the simple "run clear() with specific args/env" doesn't work since `--embed` skips startup scripts.  Screen tests felt like overkill, but that would have at least caught the memory leak I fixed in the last update of the PR.

**Comment by justinmk** [comment]
> the simple "run clear() with specific args/env" doesn't work since `--embed` skips startup scripts

@jamessan what startup scripts do you mean? `--embed` should not skip anything. 

This loads my personal config:

    nvim --headless --embed +':call writefile([localtime()." ".($MYVIMRC)." ".execute("scriptnames")],"foo.txt","a")' +q

**Comment by jamessan** [comment]
Finally added some testing.

**Review by justinmk — APPROVED** [review]

File: runtime/doc/starting.txt [hunk 3]
```diff
@@ -436,10 +436,12 @@ accordingly.  Vim proceeds in this order:
 	If Vim was started in Ex mode with the "-s" argument, all following
 	initializations until 4. are skipped.  Only the "-u" option is
 	interpreted.
-							*system-vimrc*
+						*system-vimrc* *sysinit.vim*
      a. For Unix, MS-Windows, and Macintosh, the system vimrc file is read for
```
[the remark this answers is not in the document]
Comment (jamessan) [reply runtime/doc/starting.txt]: Noted.  `:version` and `nvim --version` should probably also mention `$XDG_CONFIG_DIRS/nvim/sysinit.vim`.

[the remark this answers is not in the document]
Comment (jamessan) [reply runtime/doc/starting.txt]: Changed the doc, but I've left off the `:version`/`--version` tweaks for now.

**Review by justinmk — APPROVED** [review]

**Pull request merged**